maths/factors.py

```diff
@@ -9,7 +12,21 @@ def factors_of_a_number(num: int) -> list:
     >>> factors_of_a_number(-24)
     []
     """
-    return [i for i in range(1, num + 1) if num % i == 0]
+    facs: list[int] = []
+    if num < 1:
+        return facs
+    facs.append(1)
+    if num == 1:
+        return facs
+    facs.append(num)
+    for i in range(2, int(sqrt(num)) + 1):
+        if num % i == 0:  # If i is a factor of num
+            facs.append(i)
+            d = num // i  # num//i is the other factor of num
+            if d != i:  # If d and i are distinct
+                facs.append(d)  # we have found another factor
+    facs.sort()
```

Comment:
The `sort` function means that the time complexity becomes `O(n sqrt(n))`

Reply:
> It would actually be O(sqrt(n) * log(sqrt(n))) since the length of the list is <= 2*sqrt(n)

I see

Reply:
@CaedenPH requesting you to please approve the review so the PR may be merged. Thanks